Failed test: Test_NestedModules

### Steps to reproduce

run functional tests

### Observed behavior

```
=== FAIL: test/functional/shared/mechanics Test_NestedModules/deploy_testdata/corerp-mechanics-nestedmodules.bicep (16.80s)
    rptest.go:272: running step 0 of 1: deploy testdata/corerp-mechanics-nestedmodules.bicep
    deployexecutor.go:80: deploying deploy testdata/corerp-mechanics-nestedmodules.bicep from file testdata/corerp-mechanics-nestedmodules.bicep
2024/02/21 08:42:04 Start streaming Kubernetes logs - Pod ucp-58766c44dd-9zv2n is in state: Running
2024/02/21 08:42:04 Streaming Kubernetes logs to /home/runner/work/radius/radius/dist/container_logs/ucp-58766c44dd-9zv2n.ucp.log
    cli.go:4[71](https://github.com/radius-project/radius/actions/runs/7986125516/job/21806082291#step:26:72): [heartbeat] command rad deploy /home/runner/work/radius/radius/test/functional/shared/mechanics/testdata/corerp-mechanics-nestedmodules.bicep is still running after 10.004037936s
make: *** [build/test.mk:70: test-functional-shared] Error 1
    cli.go:418: [rad] Building /home/runner/work/radius/radius/test/functional/shared/mechanics/testdata/corerp-mechanics-nestedmodules.bicep...
    cli.go:418: [rad] Error: unexpected end of JSON input
    cli.go:418: [rad] 
    cli.go:418: [rad] TraceId:  2c98902e5d7db9b7fb9644decb789bb0
    cli.go:418: [rad] 
    cli.go:418: [rad] 
    deployexecutor.go:83: 
        	Error Trace:	/home/runner/work/radius/radius/test/step/deployexecutor.go:83
        	            				/home/runner/work/radius/radius/test/functional/shared/rptest.go:2[73](https://github.com/radius-project/radius/actions/runs/7986125516/job/21806082291#step:26:74)
        	Error:      	Received unexpected error:
        	            	command 'rad deploy /home/runner/work/radius/radius/test/functional/shared/mechanics/testdata/corerp-mechanics-nestedmodules.bicep' had non-zero exit code: exit status 1
        	Test:       	Test_NestedModules/deploy_testdata/corerp-mechanics-nestedmodules.bicep
        	Messages:   	failed to deploy deploy testdata/corerp-mechanics-nestedmodules.bicep
    --- FAIL: Test_NestedModules/deploy_testdata/corerp-mechanics-nestedmodules.bicep (16.[80](https://github.com/radius-project/radius/actions/runs/7986125516/job/21806082291#step:26:81)s)
```

### Desired behavior

tests should pass

### Workaround

_No response_

### rad Version

functional tests

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [X] Yes, I would like to support you

[AB#11266](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11266)

--- a/pkg/cli/bicep/build.go
+++ b/pkg/cli/bicep/build.go
@@ -59,12 +59,10 @@ func runBicepRaw(args ...string) ([]byte, error) {
 		return nil, fmt.Errorf("failed executing %q: %w", fullCmd, err)
 	}
 
-	// asynchronously copy to our buffer, we don't really need to observe
+	// copy to our buffer, we don't really need to observe
 	// errors here since it's copying into memory
 	buf := bytes.Buffer{}
-	go func() {
-		_, _ = io.Copy(&buf, stdout)
-	}()
+	_, _ = io.Copy(&buf, stdout)
 
 	// Wait() will wait for us to finish draining stderr before returning the exit code
 	err = c.Wait()

